typescript/infra/scripts/warp-routes/monitor-warp-routes-balances.ts

```diff
@@ -468,27 +563,89 @@ const getXERC20Limit = async (
   };
 };
 
+async function getTokenPriceByChain(
+  chain: ChainName,
+  tokenPriceGetter: CoinGeckoTokenPriceGetter,
+): Promise<number | undefined> {
+  try {
+    return tokenPriceGetter.getTokenPrice(chain);
+  } catch (e) {
+    logger.warn('Error getting token price', e);
+    return undefined;
+  }
+}
+
+async function getNativeTokenValue(
+  chain: ChainName,
+  balanceFloat: number,
+  tokenPriceGetter: CoinGeckoTokenPriceGetter,
+): Promise<number | undefined> {
+  const price = await getTokenPriceByChain(chain, tokenPriceGetter);
+  if (!price) return undefined;
+  return balanceFloat * price;
+}
+
+async function getCollateralTokenPrice(
+  tokenCoinGeckoId: string | undefined,
+  tokenPriceGetter: CoinGeckoTokenPriceGetter,
+): Promise<number | undefined> {
+  if (!tokenCoinGeckoId) return undefined;
+  const prices = await tokenPriceGetter.getTokenPriceByIds([tokenCoinGeckoId]);
+  if (!prices) return undefined;
+  return prices[0];
+}
+
+async function getCollateralTokenValue(
+  tokenCoinGeckoId: string | undefined,
+  balanceFloat: number,
+  tokenPriceGetter: CoinGeckoTokenPriceGetter,
+): Promise<number | undefined> {
+  const price = await getCollateralTokenPrice(
+    tokenCoinGeckoId,
+    tokenPriceGetter,
+  );
+  if (!price) return undefined;
+  return balanceFloat * price;
+}
+
 async function checkWarpRouteMetrics(
   checkFrequency: number,
   tokenConfig: WarpRouteConfig,
   chainMetadata: ChainMap<ChainMetadata>,
 ) {
+  const tokenPriceGetter = new CoinGeckoTokenPriceGetter(
+    new CoinGecko(),
```

Comment:
Yes I have but it may be due to me using a much higher sampling window locally, the service is resilient to rate limits, we could use our api key to eliminate this issue